browser/ui/webui/new_tab_page/ms_edge_protocol_message_handler.cc

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/webui/new_tab_page/ms_edge_protocol_message_handler.h"
+
+#include "base/bind.h"
+#include "base/command_line.h"
+#include "base/files/file_path.h"
+#include "base/logging.h"
+#include "base/notreached.h"
+#include "base/path_service.h"
+#include "base/task/thread_pool.h"
+#include "base/win/windows_version.h"
+#include "brave/browser/default_protocol_handler_utils_win.h"
+#include "chrome/installer/util/shell_util.h"
+
+using protocol_handler_utils::IsDefaultProtocolHandlerFor;
+using protocol_handler_utils::SetDefaultProtocolHandlerFor;
+
+namespace {
+
+constexpr wchar_t kMSEdgeProtocol[] = L"microsoft-edge";
+constexpr wchar_t kMSEdgeProtocolRegKey[] =
+    L"SOFTWARE\\Microsoft\\Windows\\Shell\\Associations\\UrlAssociations\\micro"
+    L"soft-edge";
+
+}  // namespace
+
+// static
+bool MSEdgeProtocolMessageHandler::CanSetDefaultMSEdgeProtocolHandler() {
+  base::win::OSInfo* os_info = base::win::OSInfo::GetInstance();
+  const auto& version_number = os_info->version_number();
+  // MS will not allow setting 3p application as a default microsoft-edge
+  // handler. See
+  // https://www.ctrl.blog/entry/microsoft-edge-protocol-competition.html
+  // Hope this constraint disappeared!
+  return version_number.major <= 11 && version_number.build < 22494;
```

Comment:
but what if major == 10 but build >= 22494? not sure how Windows versioning works, but this comparison looks a little off.

Reply:
Hmm,I thought build number is always increased regardless of major/minor.
 How about this?
```
  if (version_number.major <= 10)
    return true;
  return version_number.build < 22494;
```

Reply:
> Hmm,I thought build number is always increased regardless of major/minor. How about this?
> 
> ```
>   if (version_number.major <= 10)
>     return true;
>   return version_number.build < 22494;
> ```

Yep, I think this will be better.

Reply:
Updated. Thanks!